backend_new/test/unit/services/listing.service.spec.ts

```diff
@@ -0,0 +1,966 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { PrismaService } from '../../../src/services/prisma.service';
+import { ListingService } from '../../../src/services/listing.service';
+import { ListingsQueryParams } from '../../../src/dtos/listings/listings-query-params.dto';
+import { ListingOrderByKeys } from '../../../src/enums/listings/order-by-enum';
+import { OrderByEnum } from '../../../src/enums/shared/order-by-enum';
+import { ListingFilterKeys } from '../../../src/enums/listings/filter-key-enum';
+import { Compare } from '../../../src/dtos/shared/base-filter.dto';
+import { ListingFilterParams } from '../../../src/dtos/listings/listings-filter-params.dto';
+import { LanguagesEnum } from '@prisma/client';
+import { Unit } from '../../../src/dtos/units/unit-get.dto';
+import { UnitTypeSort } from '../../../src/utilities/unit-utilities';
+
+describe('Testing listing service', () => {
+  let service: ListingService;
+  let prisma: PrismaService;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [ListingService, PrismaService],
+    }).compile();
+
+    service = module.get<ListingService>(ListingService);
+    prisma = module.get<PrismaService>(PrismaService);
+  });
+
+  it('testing list() with no params', async () => {
+    prisma.listings.findMany = jest.fn().mockResolvedValue([
+      { id: 0, name: 'listing 1' },
```

Comment:
can we define this once so we don't have to repeat it, or debatably just have like a generateNumListings

Reply:
bump @YazeedLoonat